backend/drm/drm.c

```diff
@@ -1185,11 +1186,27 @@ void scan_drm_connectors(struct wlr_drm_backend *drm) {
 			drm_conn->encoder_id);
 
 		ssize_t index = -1;
-		struct wlr_drm_connector *c, *wlr_conn = NULL;
-		wl_list_for_each(c, &drm->outputs, link) {
+		struct wlr_drm_connector *c, *tmp, *wlr_conn = NULL;
+		size_t edid_len = 0;
+
+		uint8_t *edid = NULL;
+		wl_list_for_each_safe(c, tmp, &drm->outputs, link) {
 			index++;
 			if (c->id == drm_conn->connector_id) {
-				wlr_conn = c;
+				union wlr_drm_connector_props drm_props;
+				struct wlr_output drm_output;
+				get_drm_connector_props(drm->fd, drm_conn->connector_id, &drm_props);
+				edid = get_drm_prop_blob(drm->fd,
+					drm_conn->connector_id, drm_props.edid, &edid_len);
+				parse_edid(&drm_output, edid_len, edid);
+				if (strcmp(wlr_conn->output.make, drm_output.make) == 0 &&
```

Comment:
Instead of comparing the make/model/serial, I'd compare the blob ID. Some manufacturers don't fill make/model/serial correctly: two different screens (ie. different modes, size, etc) can have the same make/model/serial.